### PR TITLE
i18n(fr): update `sharing-state.mdx` translation

### PR DESCRIPTION
**Pull request by McFlyPartages**
#### What kind of changes does this PR include?

- Update Translated content

#### Description

- Update translation sharing-state.mdx in french

### CODE REVIEW

File: src/content/docs/fr/core-concepts/sharing-state.mdx [hunk 1]
```diff
@@ -1,40 +1,42 @@
...
 
-Les frameworks d'interface utilisateur comme React ou Vue peuvent encourager l'utilisation de [fournisseurs de "contexte"](https://fr.reactjs.org/docs/context.html) pour que d'autres composants puissent les consommer. Mais lorsque vous [hydratez partiellement des composants](/fr/core-concepts/framework-components/#hydratation-des-composants-interactifs) dans Astro ou Markdown, vous ne pouvez pas utiliser ces enveloppes contextuelles.
+Les frameworks d'interface utilisateur tels que React ou Vue peuvent encourager l'utilisation de [fournisseurs de "contexte"](https://fr.reactjs.org/docs/context.html) pour que d'autres composants puissent les consommer. Mais lorsque vous [hydratez partiellement des composants](/fr/core-concepts/framework-components/#hydratation-des-composants-interactifs) dans Astro ou Markdown, vous ne pouvez pas utiliser ces enveloppes contextuelles.
 
 Astro recommande une solution différente pour le stockage partagé côté client : les [**Nano Stores**](https://github.com/nanostores/nanostores).
 
-## Pourquoi Nano Stores&nbsp;?
+<RecipeLinks slugs={["en/recipes/sharing-state"]} />
```
Comment (pioupia) [inline src/content/docs/fr/core-concepts/sharing-state.mdx:19]: CI fix:
```suggestion
<RecipeLinks slugs={["fr/recipes/sharing-state"]} />
```

File: src/content/docs/fr/core-concepts/sharing-state.mdx [hunk 2]
```diff
@@ -43,19 +45,19 @@ Si vous comprenez ces restrictions et que vous trouvez toujours un cas d'utilisa
...
+<summary>**🙋 En quoi les Stores de Svelte sont-ils différents de Nano Stores ?**</summary>
 
-**Nano Stores et les [Stores de Svelte](https://svelte.dev/tutorial/writable-stores) sont très similaires&nbsp;!** En effet, [Nano Stores vous permettent d'utiliser le même raccourci `$`](https://github.com/nanostores/nanostores#svelte) pour les abonnements que vous pourriez utiliser avec Svelte stores.
+**Nano Stores et les [Svelte Stores](https://svelte.dev/tutorial/writable-stores) sont très similaires !** En effet, [Nano Stores vous permettent d'utiliser le même raccourci `$`](https://github.com/nanostores/nanostores#svelte) pour les abonnements que vous pourriez utiliser avec Svelte stores.
 
-Si vous voulez éviter les bibliothèques tierces, les [Stores de Svelte](https://svelte.dev/tutorial/writable-stores) sont à eux seuls un excellent outil de communication inter-composants. Néanmoins, vous préférerez peut-être Nano Stores si :
+Si vous voulez éviter les bibliothèques tierces, les [Svelte Stores](https://svelte.dev/tutorial/writable-stores) constituent à eux seuls un excellent outil de communication inter-îles. Néanmoins, vous pouvez préférerez Nano Stores si :
```
Comment (pioupia) [inline src/content/docs/fr/core-concepts/sharing-state.mdx:52]:
```suggestion
Si vous voulez éviter les bibliothèques tierces, les [Svelte Stores](https://svelte.dev/tutorial/writable-stores) constituent à eux seuls un excellent outil de communication inter-îles. Néanmoins, vous pouvez préférerer Nano Stores si :
```

File: src/content/docs/fr/core-concepts/sharing-state.mdx [hunk 3]
```diff
@@ -43,19 +45,19 @@ Si vous comprenez ces restrictions et que vous trouvez toujours un cas d'utilisa
...
 2. Ou si vous souhaitez communiquer entre Svelte et d'autres frameworks d'interface utilisateur comme Preact ou Vue.
 </details>
 
 <details>
-<summary>**🙋 Comment Solid signals est-il différent de Nano Stores&nbsp;?**</summary>
+<summary>**🙋 Comment Solid signals est-il différent de Nano Stores ?**</summary>
 
-Si vous utilisez Solid depuis un certain temps, vous avez peut-être essayé de déplacer les [signals](https://www.solidjs.com/docs/latest#createsignal) ou [stores](https://www.solidjs.com/docs/latest#createstore) en dehors de vos composants. C'est un excellent moyen de partager l'état entre les Îles de composants Solid ! Essayez d'exporter des "signals" à partir d'un fichier partagé :
+Si vous utilisez Solid depuis un certain temps, vous avez peut-être essayé de déplacer les ["signals"](https://www.solidjs.com/docs/latest#createsignal) ou [stores](https://www.solidjs.com/docs/latest#createstore) en dehors de vos composants. C'est un excellent moyen de partager l'état entre les Îles de composants Solid ! Essayez d'exporter des "signals" à partir d'un fichier partagé :
```
Comment (pioupia) [inline src/content/docs/fr/core-concepts/sharing-state.mdx:60]:
```suggestion
Si vous utilisez Solid depuis un certain temps, vous avez peut-être essayé de déplacer les ["signals"](https://www.solidjs.com/docs/latest#createsignal) ou [stores](https://www.solidjs.com/docs/latest#createstore) en dehors de vos composants. C'est un excellent moyen de partager l'état entre les îles de composants Solid ! Essayez d'exporter des "signals" à partir d'un fichier partagé :
```

File: src/content/docs/fr/core-concepts/sharing-state.mdx [hunk 4]
```diff
@@ -134,7 +136,7 @@ import AddToCartForm from '../components/AddToCartForm';
 <body>
   <header>
     <nav>
-      <a href="/">Vitrine Astro</a>
+      <a href="/">Astro storefront</a>
```
Comment (pioupia) [inline src/content/docs/fr/core-concepts/sharing-state.mdx:139]: I prefer the previous translation. You can keep this English version, but `Vitrine` do the job!
```suggestion
      <a href="/">Vitrine Astro</a>
```

File: src/content/docs/fr/core-concepts/sharing-state.mdx [hunk 5]
```diff
@@ -172,11 +174,11 @@ import { useStore } from '@nanostores/preact';
...
 
 export default function CartButton() {
-  // lire la valeur du store avec le hook `useStore`
+  // read the store value with the `useStore` hook
   const $isCartOpen = useStore(isCartOpen);
-  // écrire dans le store importé en utilisant `.set`
+  // write to the imported store using `.set`
   return (
-    <button onClick={() => isCartOpen.set(!$isCartOpen)}>Panier</button>
+    <button onClick={() => isCartOpen.set(!$isCartOpen)}>Cart</button>
```
Comment (pioupia) [inline src/content/docs/fr/core-concepts/sharing-state.mdx:181]:
```suggestion
    <button onClick={() => isCartOpen.set(!$isCartOpen)}>Panier</button>
```

File: src/content/docs/fr/core-concepts/sharing-state.mdx [hunk 6]
```diff
@@ -188,11 +190,11 @@ import { useStore } from '@nanostores/react';
 import { isCartOpen } from '../cartStore';
 
 export default function CartButton() {
-  // lire la valeur du store avec le hook `useStore`
+  // read the store value with the `useStore` hook
```
Comment (pioupia) [inline src/content/docs/fr/core-concepts/sharing-state.mdx:193]:
```suggestion
  // lire la valeur du store avec le hook `useStore`
```

File: src/content/docs/fr/core-concepts/sharing-state.mdx [hunk 7]
```diff
@@ -188,11 +190,11 @@ import { useStore } from '@nanostores/react';
...
   const $isCartOpen = useStore(isCartOpen);
-  // écrire dans le store importé en utilisant `.set`
+  // write to the imported store using `.set`
```
Comment (pioupia) [inline src/content/docs/fr/core-concepts/sharing-state.mdx:195]:
```suggestion
  // écrire dans le store importé en utilisant `.set`
```

File: src/content/docs/fr/core-concepts/sharing-state.mdx [hunk 8]
```diff
@@ -188,11 +190,11 @@ import { useStore } from '@nanostores/react';
...
   return (
-    <button onClick={() => isCartOpen.set(!$isCartOpen())}>Panier</button>
+    <button onClick={() => isCartOpen.set(!$isCartOpen)}>Cart</button>
```
Comment (pioupia) [inline src/content/docs/fr/core-concepts/sharing-state.mdx:197]:
```suggestion
    <button onClick={() => isCartOpen.set(!$isCartOpen)}>Panier</button>
```

File: src/content/docs/fr/core-concepts/sharing-state.mdx [hunk 9]
```diff
@@ -204,11 +206,11 @@ import { useStore } from '@nanostores/solid';
 import { isCartOpen } from '../cartStore';
 
 export default function CartButton() {
-  // lire la valeur du store avec le hook `useStore`
+  // read the store value with the `useStore` hook
```
Comment (pioupia) [inline src/content/docs/fr/core-concepts/sharing-state.mdx:209]:
```suggestion
  // lire la valeur du store avec le hook `useStore`
```

File: src/content/docs/fr/core-concepts/sharing-state.mdx [hunk 10]
```diff
@@ -204,11 +206,11 @@ import { useStore } from '@nanostores/solid';
...
   const $isCartOpen = useStore(isCartOpen);
-  // écrire dans le store importé en utilisant `.set`
+  // write to the imported store using `.set`
```
Comment (pioupia) [inline src/content/docs/fr/core-concepts/sharing-state.mdx:211]:
```suggestion
  // écrire dans le store importé en utilisant `.set`
```

File: src/content/docs/fr/core-concepts/sharing-state.mdx [hunk 11]
```diff
@@ -204,11 +206,11 @@ import { useStore } from '@nanostores/solid';
...
   return (
-    <button onClick={() => isCartOpen.set(!$isCartOpen)}>Panier</button>
+    <button onClick={() => isCartOpen.set(!$isCartOpen())}>Cart</button>
```
Comment (pioupia) [inline src/content/docs/fr/core-concepts/sharing-state.mdx:213]:
```suggestion
    <button onClick={() => isCartOpen.set(!$isCartOpen())}>Panier</button>
```

File: src/content/docs/fr/core-concepts/sharing-state.mdx [hunk 12]
```diff
@@ -220,27 +222,48 @@ export default function CartButton() {
   import { isCartOpen } from '../cartStore';
 </script>
 
-<!--utilisez "$" pour lire la valeur du store-->
-<button on:click={() => isCartOpen.set(!$isCartOpen)}>Panier</button>
+<!--use "$" to read the store value-->
```
Comment (pioupia) [inline src/content/docs/fr/core-concepts/sharing-state.mdx:225]:
```suggestion
<!--utilisez "$" pour lire la valeur du store-->
```

File: src/content/docs/fr/core-concepts/sharing-state.mdx [hunk 13]
```diff
@@ -220,27 +222,48 @@ export default function CartButton() {
...
+<button on:click={() => isCartOpen.set(!$isCartOpen)}>Cart</button>
```
Comment (pioupia) [inline src/content/docs/fr/core-concepts/sharing-state.mdx:226]:
```suggestion
<button on:click={() => isCartOpen.set(!$isCartOpen)}>Panier</button>
```

File: src/content/docs/fr/core-concepts/sharing-state.mdx [hunk 14]
```diff
@@ -220,27 +222,48 @@ export default function CartButton() {
...
 ```
 </Fragment>
 <Fragment slot="vue">
 ```vue
 <!--src/components/CartFlyoutToggle.vue-->
 <template>
-  <!--écrire dans le store importé en utilisant `.set`-->
-  <button @click="isCartOpen.set(!$isCartOpen)">Panier</button>
+  <!--write to the imported store using `.set`-->
```
Comment (pioupia) [inline src/content/docs/fr/core-concepts/sharing-state.mdx:233]:
```suggestion
  <!--écrire dans le store importé en utilisant `.set`-->
```

File: src/content/docs/fr/core-concepts/sharing-state.mdx [hunk 15]
```diff
@@ -220,27 +222,48 @@ export default function CartButton() {
...
+  <button @click="isCartOpen.set(!$isCartOpen)">Cart</button>
```
Comment (pioupia) [inline src/content/docs/fr/core-concepts/sharing-state.mdx:234]:
```suggestion
  <button @click="isCartOpen.set(!$isCartOpen)">Panier</button>
```

File: src/content/docs/fr/core-concepts/sharing-state.mdx [hunk 16]
```diff
@@ -220,27 +222,48 @@ export default function CartButton() {
...
 </template>
 
 <script setup>
   import { isCartOpen } from '../cartStore';
   import { useStore } from '@nanostores/vue';
 
-  // lire la valeur du store avec le hook `useStore`
+  // read the store value with the `useStore` hook
```
Comment (pioupia) [inline src/content/docs/fr/core-concepts/sharing-state.mdx:241]:
```suggestion
  // lire la valeur du store avec le hook `useStore`
```

File: src/content/docs/fr/core-concepts/sharing-state.mdx [hunk 17]
```diff
@@ -220,27 +222,48 @@ export default function CartButton() {
...
+import { isCartOpen } from '../cartStore';
+
+export class CartFlyoutToggle extends LitElement {
+  handleClick() {
+    isCartOpen.set(!isCartOpen.get());
+  }
+
+  render() {
+    return html`
+      <button @click="${this.handleClick}">Cart</button>
```
Comment (pioupia) [inline src/content/docs/fr/core-concepts/sharing-state.mdx:259]:
```suggestion
      <button @click="${this.handleClick}">Panier</button>
```

File: src/content/docs/fr/core-concepts/sharing-state.mdx [hunk 18]
```diff
@@ -430,8 +472,7 @@ Avec notre store en place, nous pouvons appeler cette fonction dans notre `AddTo
 import { addCartItem, isCartOpen } from '../cartStore';
 
 export default function AddToCartForm({ children }) {
-  // nous allons hardcoder les informations sur l'article pour plus de simplicité !
-
+  // we'll hardcode the item info for simplicity!
```
Comment (pioupia) [inline src/content/docs/fr/core-concepts/sharing-state.mdx:475]:
```suggestion
  // nous allons hardcoder les informations sur l'article pour plus de simplicité !
```

File: src/content/docs/fr/core-concepts/sharing-state.mdx [hunk 19]
```diff
@@ -458,7 +499,7 @@ export default function AddToCartForm({ children }) {
 import { addCartItem, isCartOpen } from '../cartStore';
 
 export default function AddToCartForm({ children }) {
-  // nous allons hardcoder les informations sur l'article pour plus de simplicité!
+  // we'll hardcode the item info for simplicity!
```
Comment (pioupia) [inline src/content/docs/fr/core-concepts/sharing-state.mdx:502]:
```suggestion
  // nous allons hardcoder les informations sur l'article pour plus de simplicité !
```

File: src/content/docs/fr/core-concepts/sharing-state.mdx [hunk 20]
```diff
@@ -485,7 +526,7 @@ export default function AddToCartForm({ children }) {
...
```
Comment (pioupia) [inline src/content/docs/fr/core-concepts/sharing-state.mdx:529]:
```suggestion
  // nous allons hardcoder les informations sur l'article pour plus de simplicité !
```

File: src/content/docs/fr/core-concepts/sharing-state.mdx [hunk 21]
```diff
@@ -516,7 +557,7 @@ export default function AddToCartForm({ children }) {
 <script>
   import { addCartItem, isCartOpen } from '../cartStore';
 
-  // nous allons hardcoder les informations sur l'article pour plus de simplicité!
+  // we'll hardcode the item info for simplicity!
```
Comment (pioupia) [inline src/content/docs/fr/core-concepts/sharing-state.mdx:560]:
```suggestion
  // nous allons hardcoder les informations sur l'article pour plus de simplicité !
```

File: src/content/docs/fr/core-concepts/sharing-state.mdx [hunk 22]
```diff
@@ -542,7 +583,7 @@ export default function AddToCartForm({ children }) {
 <script setup>
   import { addCartItem, isCartOpen } from '../cartStore';
 
-  // nous allons hardcoder les informations sur l'article pour plus de simplicité!
+  // we'll hardcode the item info for simplicity!
```
Comment (pioupia) [inline src/content/docs/fr/core-concepts/sharing-state.mdx:586]:
```suggestion
  // nous allons hardcoder les informations sur l'article pour plus de simplicité !
```

File: src/content/docs/fr/core-concepts/sharing-state.mdx [hunk 23]
```diff
@@ -580,11 +656,11 @@ export default function CartFlyout() {
             <li>
               <img src={cartItem.imageSrc} alt={cartItem.name} />
               <h3>{cartItem.name}</h3>
-              <p>Quantité : {cartItem.quantity}</p>
+              <p>Quantity: {cartItem.quantity}</p>
```
Comment (pioupia) [inline src/content/docs/fr/core-concepts/sharing-state.mdx:659]:
```suggestion
              <p>Quantité : {cartItem.quantity}</p>
```

File: src/content/docs/fr/core-concepts/sharing-state.mdx [hunk 24]
```diff
@@ -580,11 +656,11 @@ export default function CartFlyout() {
...
             </li>
           ))}
         </ul>
-      ) : <p>Votre panier est vide !</p>}
+      ) : <p>Your cart is empty!</p>}
```
Comment (pioupia) [inline src/content/docs/fr/core-concepts/sharing-state.mdx:663]:
```suggestion
      ) : <p>Votre panier est vide !</p>}
```

File: src/content/docs/fr/core-concepts/sharing-state.mdx [hunk 25]
```diff
@@ -608,11 +684,11 @@ export default function CartFlyout() {
             <li>
               <img src={cartItem.imageSrc} alt={cartItem.name} />
               <h3>{cartItem.name}</h3>
-              <p>Quantité : {cartItem.quantity}</p>
+              <p>Quantity: {cartItem.quantity}</p>
```
Comment (pioupia) [inline src/content/docs/fr/core-concepts/sharing-state.mdx:687]:
```suggestion
              <p>Quantité : {cartItem.quantity}</p>
```

File: src/content/docs/fr/core-concepts/sharing-state.mdx [hunk 26]
```diff
@@ -608,11 +684,11 @@ export default function CartFlyout() {
...
             </li>
           ))}
         </ul>
-      ) : <p>Votre panier est vide !</p>}
+      ) : <p>Your cart is empty!</p>}
```
Comment (pioupia) [inline src/content/docs/fr/core-concepts/sharing-state.mdx:691]:
```suggestion
      ) : <p>Votre panier est vide !</p>}
```

File: src/content/docs/fr/core-concepts/sharing-state.mdx [hunk 27]
```diff
@@ -636,11 +712,11 @@ export default function CartFlyout() {
             <li>
               <img src={cartItem.imageSrc} alt={cartItem.name} />
               <h3>{cartItem.name}</h3>
-              <p>Quantité : {cartItem.quantity}</p>
+              <p>Quantity: {cartItem.quantity}</p>
```
Comment (pioupia) [inline src/content/docs/fr/core-concepts/sharing-state.mdx:715]:
```suggestion
              <p>Quantité : {cartItem.quantity}</p>
```

File: src/content/docs/fr/core-concepts/sharing-state.mdx [hunk 28]
```diff
@@ -660,12 +736,12 @@ export default function CartFlyout() {
...
         <img src={cartItem.imageSrc} alt={cartItem.name} />
         <h3>{cartItem.name}</h3>
-        <p>Quantité : {cartItem.quantity}</p>
+        <p>Quantity: {cartItem.quantity}</p>
       </li>
       {/each}
     </aside>
   {#else}
-    <p>Votre panier est vide !</p>
+    <p>Your cart is empty!</p>
```
Comment (pioupia) [inline src/content/docs/fr/core-concepts/sharing-state.mdx:744]:
```suggestion
    <p>Votre panier est vide !</p>
```

File: src/content/docs/fr/core-concepts/sharing-state.mdx [hunk 29]
```diff
@@ -679,10 +755,10 @@ export default function CartFlyout() {
       <li v-for="cartItem in Object.values($cartItems)" v-bind:key="cartItem.name">
         <img :src=cartItem.imageSrc :alt=cartItem.name />
         <h3>{{cartItem.name}}</h3>
-        <p>Quantité : {{cartItem.quantity}}</p>
+        <p>Quantity: {{cartItem.quantity}}</p>
```
Comment (pioupia) [inline src/content/docs/fr/core-concepts/sharing-state.mdx:758]:
```suggestion
        <p>Quantité : {{cartItem.quantity}}</p>
```

File: src/content/docs/fr/core-concepts/sharing-state.mdx [hunk 30]
```diff
@@ -679,10 +755,10 @@ export default function CartFlyout() {
...
       </li>
     </ul>
-    <p v-else>Votre panier est vide !</p>
+    <p v-else>Your cart is empty!</p>
```
Comment (pioupia) [inline src/content/docs/fr/core-concepts/sharing-state.mdx:761]:
```suggestion
    <p v-else>Votre panier est vide !</p>
```

File: src/content/docs/fr/core-concepts/sharing-state.mdx [hunk 31]
```diff
@@ -695,9 +771,54 @@ export default function CartFlyout() {
...
+export class CartFlyoutLit extends LitElement {
+  private cartOpen = new StoreController(this, isCartOpen);
+  private getCartItems = new StoreController(this, cartItems);
+
+  renderCartItem(cartItem) {
+    return html`
+      <li>
+        <img src="${cartItem.imageSrc}" alt="${cartItem.name}" />
+        <h3>${cartItem.name}</h3>
+        <p>Quantity: ${cartItem.quantity}</p>
```
Comment (pioupia) [inline src/content/docs/fr/core-concepts/sharing-state.mdx:790]:
```suggestion
        <p>Quantité : ${cartItem.quantity}</p>
```

File: src/content/docs/fr/core-concepts/sharing-state.mdx [hunk 32]
```diff
@@ -695,9 +771,54 @@ export default function CartFlyout() {
...
+            ${
+              Object.values(this.getCartItems.value).length
+                ? html`
+                  <ul>
+                    ${Object.values(this.getCartItems.value).map((cartItem) =>
+                      this.renderCartItem(cartItem)
+                    )}
+                  </ul>
+                `
+                : html`<p>Your cart is empty!</p>`
```
Comment (pioupia) [inline src/content/docs/fr/core-concepts/sharing-state.mdx:808]:
```suggestion
                : html`<p>Votre panier est vide !</p>`
```

**Review by pioupia — CHANGES_REQUESTED** [review]
Some little mistakes.
I translated also all the text / comment in the code. If you don't want to translate them in the final translation just ignore these edits (its a big part of the changes ~ like about 30 of 34 requests).

The rest LGTM!

**Comment by McFlyPartages** [comment]
Thx for code translation @pioupia and this correction

**Review by pioupia — APPROVED** [review]
LGTM

**Review by Yan-Thomas — APPROVED** [review]
LGTM! Thanks peeps 🙌

**Pull request merged**